### ISSUE TITLE
Pipeline

### ISSUE THREAD

**Issue by adrn** [adrn/joaquin#9]
- Use PCA to split the stars into neighborhoods: each _neighborhood_ with size `neighborhood_size` is defined by a _model citizen_ (an actual star). The nearest `block_size` stars (in PCA feature-space) are the _block_, a subset of the neighborhood.
- for neighborhood in neighborhoods:
    - Patch the spectra with PCA: 
        - Construct a `patching_n_components` PCA of all spectra in a neighborhood, setting all masked pixels to 0.
        - Fill in masked pixels with a PCA reconstruction of each spectrum at each masked pixel
    - Apply the NUFFT low-pass filter to each spectrum
    - TODO: figure out regression hyper-parameters (L2_ivar, parallax zeropoint) or check if we can set them globally
    - For `K` in `Kfold_K`:
        - Train regression on the training data from large neighborhood (zone 2)
        - apply results to held-out and parent-not-training data in small neighborhood (zone 1)

**Comment by davidwhogg** [comment]
*Edited: moved into top comment*
